### DISCUSSION TITLE
Get commands under subcommands like wait

### DISCUSSION THREAD

**Discussion by mccgeoff** [aws/aws-cli#7708] [Q&A]
Some commands have subcommands beneath them, like wait: https://docs.aws.amazon.com/cli/latest/reference/ecs/wait/index.html. Is there a way to get these sub-subcommands? I'm able to get the top-level commands for each service (like aws ec2 wait) using the following Python, but I'm not sure how to get the sub-subcommands when they are available (like aws ec2 wait services-inactive):

```
import awscli.completer
import os.path

exampledir = os.path.dirname(__file__)+"awscli/awscli/examples/"
completer = awscli.completer.Completer()
commands = completer.complete("aws", len("aws"))
missing = []
totalcommands = 0
totalexamples = 0

def process(command):
  global totalcommands
  global totalexamples
  subcommands = completer.complete(command, len(command))
  for subcommand in subcommands:
    print(subcommand)

  print(command.upper().ljust(30))

for command in commands:
  process(command)
```
I would have thought that `commands = completer.complete("aws ec2 wait")` in this case would have worked, but it does not.

**Comment by tim-finnigan** [comment]
Hi @mccgeoff, I think you have a couple options here. One is to use the [help command](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-help.html#cli-usage-help-command) (for example `aws ec2 wait help`) to see the available commands. Another option is to enable [auto-prompt mode](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-parameters-prompting.html). If you type in `aws ec2 wait` with auto-prompt enabled then you will see the options like this:

<img width="353" alt="image" src="https://user-images.githubusercontent.com/87778557/226053044-3b33b17a-685d-42cf-989d-a54d29c7e61a.png">

Does that accomplish what you're looking for?

**Reply by mccgeoff** [reply]
@tim-finnigan. Thank you for the suggested options. I didn't know about auto-prompt mode. While these options do work in the AWS CLI to show the optional wait commands, they don't seem to work when using completer. I've just tried them. Take the following example:
```
import awscli.completer

completer = awscli.completer.Completer()
commands = completer.complete("aws ec2")

for command in commands:
    print("command", command)
```
When you run that, you get back all of the completed "aws ec2 ..." commands:
```
command accept-address-transfer
command accept-reserved-instances-exchange-quote
command accept-transit-gateway-multicast-domain-associations
command accept-transit-gateway-peering-attachment
command accept-transit-gateway-vpc-attachment
command accept-vpc-endpoint-connections
command accept-vpc-peering-connection
command advertise-byoip-cidr
```
Now, if you change it to "aws ec2 wait help" or "aws ec2 wait --cli-auto-prompt", no values or command options are returned. So that's not working as an option.

I've found a workaround of sorts: I can just use boto3 to get the wait operation names back, but they come back formatted with underscores (like 'bundle_task_complete'). I can then convert them to dashes (like 'bundle-task-complete') and then I have a list of wait operations, but it doesn't feel like a very elegant way to do it.
